src/hooks/useRecommendationsObserver.ts

```diff
@@ -0,0 +1,57 @@
+import { useEffect } from 'react';
+import { Nullable } from '@constructor-io/constructorio-client-javascript/lib/types';
+import ConstructorIO from '@constructor-io/constructorio-client-javascript';
+import { Section } from '../types';
+
+function useRecommendationsObserver(
+  menuIsOpen: boolean,
+  sections: Section[],
+  constructorIO: Nullable<ConstructorIO>,
+  callback: (
+    target: HTMLElement,
+    sections: Section[],
+    constructorIO: Nullable<ConstructorIO>
+  ) => void
+) {
+  // Get refs for each section
+  const refs = sections
+    .filter((section) => section.type === 'recommendations')
+    .map((section) => section.ref);
+
+  useEffect(() => {
+    const observer = new IntersectionObserver(
```

Comment:
I'm confused. What do we mean by "intersection" here?

Reply:
whoa... https://developer.mozilla.org/en-US/docs/Web/API/Intersection_Observer_API

this is very coooool :)